### PR TITLE
Bens branch mobile

### PR DESCRIPTION
**Pull request by bdesollar**
<!Pull Request Template>

## Description

This pull request introduces several significant updates to the `Rewards.js` and `Tasks.js` components, along with corresponding backend adjustments in `TaskRoutes.js`. The primary focus was on enhancing the functionality of the rewards page, completing the implementation of the tasks feature, and resolving existing bugs in task management routes.

### Key Changes:
- Customized the `Rewards.js` page to provide a more engaging user experience through animations and a dynamic display of the user's total points. As well as scroll through the completed tasks.
- Completed the `Tasks.js` component, ensuring tasks can be fetched, displayed, and updated correctly. This includes the ability to mark tasks as completed and the integration of sound playback for task notifications.
- Updated `TaskRoutes.js` on the server-side to fix bugs related to task creation and updates, improving the reliability of task management.
- Added new libraries: `react-native-confetti-cannon` for the confetti animation in the `Rewards.js` page and `expo-av` for handling audio playback in the `Tasks.js` component.

## Tests

Added comprehensive unit tests for both the `Tasks.js` and `Rewards.js` components to ensure functionality works as expected and to maintain high code quality. The tests cover:
- Fetching and displaying tasks and rewards correctly.
- Interactions within the components, such as marking tasks as completed.
- Error handling and edge cases, ensuring robustness.

## UI Changes

- The `Rewards.js` page now features a confetti animation to celebrate the user's achievements, enhancing visual feedback and user engagement.
- The `Tasks.js` component has been updated to include visual and audio notifications for tasks, making the task management experience more interactive.
image
image

## Issues
#99 #117 #116 #102 #112 #111 #118 #120 

## Additional Notes

- The introduction of audio playback for tasks required the addition of the `expo-av` library. Ensure to install this dependency when pulling the changes.
- The confetti animation in the `Rewards.js` page is implemented using the `react-native-confetti-cannon` library, which was added to enhance the user experience.

---

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (bdesollar) [inline .pre-commit-config.yaml:1]: Had to remove this to upload the mp3 sounds

File: server/routes/AuthRoute.js [hunk 1]
```diff
@@ -4,31 +4,42 @@ import jwt from 'jsonwebtoken';
...
   const { username, password } = req.body;
+  // Find the user by username
   const user = await UserModel.findOne({ username });
   if (!user) {
-    return res.json({ message: 'User does not exist' });
+    return res.status(404).json({ message: 'User does not exist' }); // Use HTTP status code 404 for not found
   }
+  // Compare the provided password with the hashed password in the database
   const isPasswordValid = await bcrypt.compare(password, user.password);
+  console.log('isPasswordValid: ', isPasswordValid);
```
Comment (saanbe16) [inline server/routes/AuthRoute.js:36]: Could you remove this log

File: server/routes/TaskRoutes.js [hunk 2]
```diff
@@ -5,11 +5,14 @@ const router = express.Router();
 
 // Create task route
 router.post('/create', async (req, res) => {
+  console.log('Creating task for user:');
```
Comment (saanbe16) [inline server/routes/TaskRoutes.js:8]: This too

File: server/routes/TaskRoutes.js [hunk 3]
```diff
@@ -5,11 +5,14 @@ const router = express.Router();
...
   const { username, taskType, date, startTime, endTime, duration, points } = req.body;
+  console.log('Creating task for user with these details:', req.body);
```
Comment (saanbe16) [inline server/routes/TaskRoutes.js:10]: and here

File: server/routes/TaskRoutes.js [hunk 4]
```diff
@@ -5,11 +5,14 @@ const router = express.Router();
...
   const user = await UserModel.findOne({ username });
   if (!user) {
-    return res.json({ message: 'Invalid user' });
+    return res.status(401).json({ message: 'Invalid user' });
   }
+  console.log('Creating task for user:', user);
```
Comment (saanbe16) [inline server/routes/TaskRoutes.js:15]: Can you either remove this or change it to be user.username or something to make the output less verbose

**Review by saanbe16 — CHANGES_REQUESTED** [review]
@bdesollar this looks great, could you go through the server and remove some of the logging statements I commented on though? They cause the tests to get extremely verbose really quickly and make debugging them a lot more annoying.

File: server/routes/AuthRoute.js (hunk 1 above)
Comment (bdesollar) [reply server/routes/AuthRoute.js:36]: Done

Comment (bdesollar) [reply server/routes/AuthRoute.js]: Done

**Review by saanbe16 — APPROVED** [review]
